Enterprise-CMCS/macpro-mdct-mfp
# CMDCT-159 Update due date calculation

### Description

This change 
- adds a method to the create API to calculate the due date for a given **Work Plan** or **SAR** report
- removes the **calculateDueDate** method from the `ui-src` `time` file
- adds tests to verify if a given year is a **leap year** 
- adds tests to verify the due date is calculate correctly for a **Work Plan** or a **SAR** report created in **Period 1** or **Period 2**

### Related ticket(s)


CMDCT-159

Due Date Logic: 
 * WP due date is May 1 for Period 1, and Nov 1 for Period 2
 * SAR due date for Period 1{*}: 60 days from June 30, due date is Aug 29.
 * SAR due date for Period 2{*}: 60 days from December 31(for a non leap year Mar 1, for leap years it’s Feb 29).
 
---
### How to test

1. login as state user
2. create work plan
3. check that the due date is correct


https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/9246062/023e2859-5865-43c7-b28c-2c55f494a8f0

Note: I am not sure why my SAR report looks a bit jank and doesn't have a due date on the frontend. Perhaps it is because the WP is not actually completed? Happy to put up more commits if I need to change something to fix this


UPDATED VIDEO!! 😄

https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/9246062/874e012c-c7ac-472c-9230-be3805906a25



### Important updates







---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
